client/state/plugins/installed/selectors-ts.ts

```diff
@@ -95,3 +95,51 @@ export const getAllPluginsIndexedBySiteId = createSelector(
 	},
 	( state ) => [ getAllPluginsIndexedByPluginSlug( state ), getSiteIdsThatHavePlugins( state ) ]
 ) as { ( state: AppState ): { [ siteId: number ]: { [ pluginSlug: string ]: Plugin } } };
+
+export function getPluginsOnSites( state: AppState, plugins: Plugin[] ) {
+	return plugins.reduce( ( acc: { [ pluginSlug: string ]: Plugin }, plugin: Plugin ) => {
+		const siteIds = Object.keys( plugin.sites ).map( Number );
+		const pluginOnSites = getPluginOnSites( state, siteIds, plugin.slug );
+		if ( pluginOnSites ) {
+			acc[ plugin.slug ] = pluginOnSites;
+		}
+		return acc;
+	}, {} );
+}
+
+export function getPluginOnSites( state: AppState, siteIds: number[], pluginSlug: string ) {
+	const plugin = getAllPluginsIndexedByPluginSlug( state )[ pluginSlug ];
+
+	if ( ! plugin ) {
+		return undefined;
+	}
+
+	if ( siteIds.some( ( siteId ) => !! plugin.sites[ siteId ] ) ) {
+		return plugin;
+	}
+
+	return undefined;
+}
+
+export function getPluginOnSite( state: AppState, siteId: number, pluginSlug: string ) {
+	const plugin = getAllPluginsIndexedByPluginSlug( state )[ pluginSlug ];
+
+	const { sites, ...pluginWithoutSites } = plugin;
+
+	if ( ! plugin || ! plugin.sites[ siteId ] ) {
+		return undefined;
+	}
+
+	// To keep compatibility with some behavior that existed before the refactor
+	// in #73296 the returned object has the site specific data lifted onto it, and
+	// the sites property has only the site data for the requested site.
+	return {
```

Comment:
This will always return a new object, regardless of whether there are differences in the data. Memoize?